File: ago-dic-2018/slides/basic/insertion.go
package main

import "fmt"

// start insertionSort OMIT
func insertionSort(arr []int) []int {
	for j := 1; j < len(arr); j++ {
		actual := arr[j]

		i := j - 1
		for i >= 0 && arr[i] > actual {
			arr[i+1] = arr[i]
			i--
		}
		arr[i+1] = actual
	}
	return arr
}

// end insertionSort OMIT

// start main OMIT
func main() {
	arr := []int{4, 2, 5, 6, 2, 4, 5, 9, 1, 3}
	fmt.Println(arr)
	sortedArr := insertionSort(arr)
	fmt.Println(sortedArr)
}

// end main OMIT

File: ene-jun-2021/material/algoritmos/insertion.java
public class insertion{
    public static void main(String [] args) {
        char []arr = {'f', 'h', 'g', 'q', 'a', 'n', 'p'};


        // Insertion sort
        for(int i = 1 ; i < arr.length ; i++){
            char actual = arr[i];
            int j = i-1;
            for(; j >= 0 ; j--){
                if(arr[j] > actual){
                    break;
                } 
                arr[j+1] = arr[j];
            }

            arr[j+1] = actual;
        }

        // Print
        for(int val:arr){
            System.out.printf("%c ", val);
        }
    }
}